### PR TITLE
[Merged by Bors] - feat: superFactorial_two_mul, superFactorial_four_mul

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Factorial/SuperFactorial.lean
Comment (mo271) [inline Mathlib/Data/Nat/Factorial/SuperFactorial.lean:128]: I'd be happy to outsource these two lemmas, see my question here: 
https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/group.20sum.2Fprod.20in.20pairs.20even.20and.20odd.20summands

File: Mathlib/Data/Nat/Factorial/SuperFactorial.lean
Comment (mo271) [reply Mathlib/Data/Nat/Factorial/SuperFactorial.lean]: .. not needed after reminding myself, there is already the same thing with `range`...

File: Mathlib/Data/Nat/Factorial/SuperFactorial.lean
Comment (urkud) [reply Mathlib/Data/Nat/Factorial/SuperFactorial.lean:122]: Why not just prove (not tested):
```lean
@[to_additive]
lemma Finset.prod_even_odd_range {M : Type*} [CommMonoid M] (n : Nat) (f : Nat → M) :
    (∏ x in range n, f (2 * x)) * (∏ x in range n, f (2 * x + 1)) = ∏ x in range (2 * n), f x :=
```
by induction on `n`?

Comment (mo271) [reply Mathlib/Data/Nat/Factorial/SuperFactorial.lean:122]: That would definitely work, see my very first message in the zulip link above. 

Do you propose to just change it to use induction and not to think about generalizing this any more?

Comment (urkud) [reply Mathlib/Data/Nat/Factorial/SuperFactorial.lean]: I pushed a proof by induction (also, it provides an explicit formula for `n`). If you want to restore helper lemmas from your old proof, then I suggest that you open a new PR with them.
I guess, we need a review from another maintainer now.
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
